maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -1380,14 +1404,14 @@ private static boolean isRunningOnCI(Properties systemProperties) {
     }
 
     private String determineLocalRepositoryPath(final MavenExecutionRequest request) {
-        String userDefinedLocalRepo = request.getUserProperties().getProperty(MavenCli.LOCAL_REPO_PROPERTY);
+        String userDefinedLocalRepo = request.getUserProperties().getProperty(Constants.MAVEN_REPO_LOCAL);
         if (userDefinedLocalRepo != null) {
             return userDefinedLocalRepo;
         }
 
         // TODO Investigate why this can also be a Java system property and not just a Maven user property like
         // other properties
-        return request.getSystemProperties().getProperty(MavenCli.LOCAL_REPO_PROPERTY);
+        return request.getSystemProperties().getProperty(Constants.MAVEN_REPO_LOCAL);
```

Comment:
@gnodet, should this also issue a warning like you did with ext classpath being a system property?